src/CSSList/CSSBlockList.php

```diff
@@ -140,4 +140,27 @@ protected function allSelectors(array &$aResult, $sSpecificitySearch = null)
             }
         }
     }
+
+    /**
+     * @param CSSList|Rule|RuleSet|Value $oElement
+     * @param array<int, CSSFunction> $aResult
+     *
+     * @return void
+     */
+    protected function allFunctions($oElement, array &$aResult)
+    {
+        if ($oElement instanceof CSSBlockList) {
+            foreach ($oElement->getContents() as $oContent) {
+                $this->allFunctions($oContent, $aResult);
+            }
+        } elseif ($oElement instanceof RuleSet) {
+            foreach ($oElement->getRules() as $oRule) {
+                $this->allFunctions($oRule, $aResult);
+            }
+        } elseif ($oElement instanceof Rule) {
+            $this->allFunctions($oElement->getValue(), $aResult);
+        } elseif ($oElement instanceof CSSFunction) {
+            $aResult[] = $oElement;
```

Comment:
Should we consider using generators to reduce memory usage? Would it be worthy 🤔

Reply:
> Should we consider using generators to reduce memory usage? Would it be worthy 🤔

Nice. Yet I don't think this would go well with upstream if this change makes it there :)  (very conservative. Maybe with good reason)